app/components/post-list.js

```diff
@@ -1,4 +1,34 @@
 import Component from '@ember/component';
+import { computed, get, set } from '@ember/object';
 
 export default Component.extend({
+
+    authors: computed('posts.@each.author', {
+        get() {
+            const array = ['All']
+            this.posts.forEach(post => {
+                if(array.indexOf(post.author) === -1 && post.author) {
+                    array.push(post.author)
+                }
+                    
+            })
+            set(this, 'author', array[0])
+            return array
+        }
+    }),
+
+    filteredPosts: computed('author', {
+        get() {
```

Comment:
@shenbaga-praburaj , no need of explicit getter setter here you can directly return the posts.